eZ/Publish/Core/Helper/ContentInfoLocationLoader/SudoMainLocationLoader.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Helper\ContentInfoLocationLoader;
+
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\Core\Helper\ContentInfoLocationLoader;
+use eZ\Publish\API\Repository\Repository;
+
+/**
+ * Loads the main location of a given ContentInfo using sudo().
+ */
+class SudoMainLocationLoader implements ContentInfoLocationLoader
+{
+    /**
+     * @var \eZ\Publish\API\Repository\Repository|\eZ\Publish\Core\Repository\Repository
+     */
+    private $repository;
+
+    public function __construct(Repository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function loadLocation(ContentInfo $contentInfo)
+    {
+        if (is_null($contentInfo->mainLocationId)) {
+            return null;
```

Comment:
Hm... It's not exceptional as far as view builder is concerned, but it is if ContentInfoLocationLoader is concerned. The loader should not concern itself with the fact that using content without location is sometimes okay.